viadot/sources/redshift_spectrum.py

```diff
@@ -0,0 +1,198 @@
+from typing import Dict, List, Literal
+
+import awswrangler as wr
+import boto3
+import pandas as pd
+
+from viadot.sources.base import Source
+
+
+class RedshiftSpectrum(Source):
+    """
+    A class for pulling data from and uploading to Spectrum.
+
+    Args:
+        profile_name (str, optional): The name of the AWS profile.
+        aws_secret_access_key (str, optional): AWS secret access key
+        aws_session_token (str, optional): AWS temporary session token
+    """
+
+    def __init__(
+        self,
+        profile_name: str = None,
+        aws_access_key_id: str = None,
+        aws_secret_access_key: str = None,
+    ):
+
+        self.profile_name = profile_name
+        self.aws_access_key_id = aws_access_key_id
+        self.aws_secret_access_key = aws_secret_access_key
+
+        self._session = None
+
+    @property
+    def session(self):
+        """A singleton-like property for initiating a session to the AWS."""
+        if not self._session:
+            self._session = boto3.session.Session(
+                profile_name=self.profile_name,
+                aws_access_key_id=self.aws_access_key_id,
+                aws_secret_access_key=self.aws_secret_access_key,
+            )
+        return self._session
+
+    def ls(
+        self,
+        database: str = None,
+        name_contains: str = None,
+        search_text: str = None,
+    ) -> List[str]:
+        """
+        Returns a list of tables in a Glue database.
+
+        Args:
+            database (str, optional): Database name.
+            name_contains (str, optional): Match by a specific substring of the table
+                name.
+            search_text (str, optional): Select only tables with the given string in
+                table's properties.
+        """
+        df = wr.catalog.tables(
+            boto3_session=self.session,
+            database=database,
+            name_contains=name_contains,
+            search_text=search_text,
+        )
+
+        return df
+
+    def exists(self, database: str, table: str) -> bool:
+        """
+        Check if a table exists in Glue database.
+
+        Args:
+            database (str): AWS Glue catalog database name.
+            table (str): AWS Glue catalog table name.
+
+        Returns:
+            bool: Whether the paths exists.
+        """
+        return wr.catalog.does_table_exist(
+            boto3_session=self.session,
+            database=database,
+            table=table,
+        )
+
+    def rm(
+        self,
+        database: str,
+        table: str,
+        remove_files: bool = True,
+    ):
+        """
+        Deletes table from AWS Glue database and related file from AWS S3, if specified.
+
+        Args:
+            path (str): Path to a folder.
+            database (str): AWS Glue catalog database name.
+            table (str): AWS Glue catalog table name.
+            remove_files (bool): If True, AWS S3 file related to the table will be removed.
```

Comment:
Missing docstring about the fact that this param is optional and that it defaults to True.

Reply:
TBD